tests/phpunit/tests/media.php

```diff
@@ -3863,6 +3863,48 @@ public function test_wp_filter_content_tags_does_not_lazy_load_images_in_header(
 		$this->assertSame( '<div class="wp-site-blocks">' . $expected_template_content . '</div>', $html );
 	}
 
+	/**
+	 * @ticket 58089
+	 *
+	 * @covers ::wp_filter_content_tags
+	 * @covers ::wp_get_loading_attr_default
+	 */
+	public function test_wp_filter_content_tags_does_not_lazy_load_special_images_within_the_content() {
+		global $wp_query, $wp_the_query;
+
+		// Force no lazy-loading on the image tag expected in the content.
+		$expected_content = wpautop( wp_get_attachment_image( self::$large_id, 'large', false, array( 'loading' => false ) ) );
+
+		// Overwrite post content with an image.
+		add_filter(
+			'the_content',
+			static function() {
+				// Replace content with an image tag, i.e. the 'wp_get_attachment_image' context is used while running 'the_content' filter.
+				return wp_get_attachment_image( self::$large_id, 'large', false );
```

Comment:
That's a great question, we should test this. I would think so, since both are static, but not 100% sure whether static closures are bound to the class they're declared in.

Reply:
Yeah this works fine. Sample test below:
```php
static $post_content = '<p>hello</p>';

public function test_static_property_in_static_closure() {
	$default_content = '<p>goodbye</p>';

	$content_before_filter = trim( apply_filters( 'the_content', $default_content ) );
	add_filter(
		'the_content',
		static function() {
			return self::$post_content;
		}
	);
	$content_after_filter = trim( apply_filters( 'the_content', $default_content ) );

	$this->assertSame(
		$default_content,
		$content_before_filter,
		'The content before filtering was incorrect.'
	);

	$this->assertSame(
		self::$post_content,
		$content_after_filter,
		'The content was not filtered correctly.'
	);
}
```

![image](https://github.com/WordPress/wordpress-develop/assets/79332690/67d5d21c-9f2b-42db-b093-08c5bec26dc4)

Reply:
That's because the closure is static, but `self::` is static too, so it makes sense that this works to me. Thanks @costdev for confirming!

Reply:
> That's because the closure is static, but self:: is static too, so it makes sense that this works to me. 

Actually, the reason it works is slightly differently. It has nothing to do with the closure being `static`.

`self::$post_content` expands at compile time to `ClassName::$post_content` and - as long as the property is `public` - is accessible like that from anywhere, which includes the closure.
See: https://3v4l.org/Lccap

Now, if the static property were `protected` or `private`, this would not work in the same way anymore.
* Protected: https://3v4l.org/up6Rc
* Private: https://3v4l.org/r9NNR

Hope that helps make sense of it.

Reply:
As long as works, I am happy.